File: 9_vowel_count/vowel_count.jl
function count_vowels(s::String)
    vowels = ['a', 'e', 'i', 'o', 'u']

    num_vowels = 0
    for c in s
        if c ∈ vowels
            num_vowels += 1
        end
    end

    num_cons = length(s) - num_vowels

    return num_vowels, num_cons
end

print("Enter a word: ")
s = readline()
num_vowels, num_cons = count_vowels(s)
print("$(s) has $(num_vowels) vowels and $(num_cons) consonants.")

File: 9_vowel_count/vowel_count.cpp
#include <iostream>

using namespace std;

struct counts {
    int num_vowels;
    int num_consonants;
};

counts count_vowels(string s) {
    char vowels[] = {'a', 'e', 'i', 'o', 'u'};

    counts out_counts;
    out_counts.num_vowels = 0;

    for (int i = 0; i <= s.length(); i++) {
        if (find(begin(vowels), end(vowels), s[i]) != end(vowels)) {
            out_counts.num_vowels++;
        };
    };

    out_counts.num_consonants  = s.length() - out_counts.num_vowels;

    return out_counts;
};

int main() {
    cout << "Enter a word: ";
    string s;
    cin >> s;
    counts res = count_vowels(s);
    cout << s << " has " << res.num_vowels << " vowels and " << res.num_consonants << " consonants.";
};